cmd/ci-secret-bootstrap/main.go

```diff
@@ -541,18 +541,31 @@ func updateSecrets(getters map[string]Getter, secretsMap map[string][]*coreapi.S
 
 			if secret.Namespace == "openshift-config" && secret.Name == "pull-secret" && osdGlobalPullSecretGroup.Has(cluster) {
 				logger.Debug("handling the global pull secret on an OSD cluster")
-				if mutated, err := mutateGlobalPullSecret(existingSecret, secret); err != nil {
+				mutated, err := mutateGlobalPullSecret(existingSecret, secret, api.DomainForService(api.ServiceRegistry))
+				if err != nil {
 					errs = append(errs, fmt.Errorf("failed to mutate secret %s:%s/%s: %w", cluster, secret.Namespace, secret.Name, err))
-				} else {
-					if mutated {
-						if _, err := secretClient.Update(context.TODO(), existingSecret, metav1.UpdateOptions{DryRun: dryRunOptions}); err != nil {
-							errs = append(errs, fmt.Errorf("error updating global pull secret %s:%s/%s: %w", cluster, existingSecret.Namespace, existingSecret.Name, err))
-						}
-						logger.Debug("global pull secret updated")
-					} else {
-						logger.Debug("global pull secret skipped")
+				}
+				mutateQCIAPPCIDomain, err := mutateGlobalPullSecret(existingSecret, secret, api.QCIAPPCIDomain)
+				if err != nil {
+					errs = append(errs, fmt.Errorf("failed to mutate secret %s:%s/%s: %w", cluster, secret.Namespace, secret.Name, err))
+				}
+				mutateQuayOpenshiftCI, err := mutateGlobalPullSecret(existingSecret, secret, api.QuayOpenShiftCIRepo)
+				if err != nil {
+					errs = append(errs, fmt.Errorf("failed to mutate secret %s:%s/%s: %w", cluster, secret.Namespace, secret.Name, err))
+				}
+				mutateQCICache, err := mutateGlobalPullSecret(existingSecret, secret, api.QCICache)
+				if err != nil {
+					errs = append(errs, fmt.Errorf("failed to mutate secret %s:%s/%s: %w", cluster, secret.Namespace, secret.Name, err))
+				}
+				if mutated || mutateQCIAPPCIDomain || mutateQuayOpenshiftCI || mutateQCICache {
```

Comment:
@smg247 I am skeptical about this statement, other than that everything else should be good